src/tools/tidy/src/walk.rs

```diff
@@ -31,6 +32,17 @@ pub fn filter_dirs(path: &Path) -> bool {
         "src/bootstrap/target",
         "vendor",
     ];
+    let command =
+        Command::new("git").args(["ls-files", ".", "--exclude-standard", "--others"]).output();
```

Comment:
This won't work properly if tidy is run from a subdirectory. Maybe this could be combined with `git rev-parse --show-toplevel`?

Reply:
Also, `ls-files` will return each untracked file in a directory rather than the directory itself. I think `--directory` is the flag you want to prevent recursing into untracked dirs

Reply:
This was the command that was posted in the linked issue, so I wasn't sure. I'll look into it.